src/service/resources-impl.js

```diff
@@ -435,6 +435,69 @@ export class Resources {
   }
 
   /**
+   * Runs the specified subtree "show" mutation and ensures that measures
+   * and layouts performed for the affected elements.
+   * @param {!Element} root
+   * @param {function()} mutator
+   * @return {!Promise}
+   */
+  showSubtree(root, mutator) {
```

Comment:
Yeah. That might be better. I was erring away from it because it's sort of annoying to do a lot of work if I only hide a small element such as a notification div. But you are probably right - the cost is probably not very high.